src/segment-loader.js

```diff
@@ -542,7 +542,11 @@ export default class SegmentLoader extends videojs.EventTarget {
     // decrease in network performance or a server issue.
 
     // don't timeout if we are on the last un-blacklisted playlist
-    segmentInfo.dontTimeout ? requestTimeout = 0 : requestTimeout = (segment.duration * 1.5) * 1000;
+    if (segmentInfo.dontTimeout) {
+      requestTimeout = 0;
+    } else {
+      requestTimeout = (segment.duration * 1.5) * 1000;
+    }
```

Comment:
Sorry, I didn't fully understand you I guess. One problem I see with that is the request timeout isn't really calculated until we know the segment duration, which is only in the segment loader. Since we won't know the timeout request time within the MPC, I think there are only 2 ways to approach passing it in. We could either pass in an enable/disable timeout property, kind of like I used above. We then could use this switch to decide what timeout to use. as an alternative we could use the goal segment duration as the calculator for the timeout request, since I believe the playlists belonging to the MPC know this. Any preference?